### PR TITLE
Don't truncate topic data unless fully replicated

### PR DESCRIPTION
**Pull request by otoolep**
Fix issue #2521

### CODE REVIEW

File: messaging/broker.go [hunk 1]
```diff
@@ -950,13 +951,19 @@ func (t *Topic) Truncate(maxSize int64) (int, int64, error) {
...
+		replicatedIndex = math.MaxUint64
+		for _, idx := range t.indexByURL {
+			if idx < replicatedIndex {
+				replicatedIndex = idx
+			}
 		}
+	} else {
+		// No nodes are replicating this topic, so it's not replicated at all.
+		replicatedIndex = 0
 	}
```
Comment (benbjohnson) [inline messaging/broker.go]: You don't need to set it since it defaults to `0` on declaration.

**Comment by benbjohnson** [comment]
I'd say remove the `else` block mentioned and then :ship:

File: messaging/broker.go (hunk 1 above)
Comment (otoolep) [inline messaging/broker.go]: Ha ha. Of course. D'oh. :-)

**Comment by otoolep** [comment]
I had sort of brain freeze and reverted to C programming. Code changed as per @benbjohnson.

Merging on green.

**Pull request merged**